Smds

hamediut opened the pull request:
I added three scripts to do the smd calculations:

1) ROI_selection_smd.py: 
it starts by going through the core image folders and finds ROI images of size in each segmented image.
it assumes that core images are already segmented. It also assumes that all the core folders from "CS_5057_5_B_1_1_1" to "CS_5057_5_B_142_4_1" are inside a parent folder.

This code saves the crop images (with the coordination of ROI in the name) in an output folder.

2) calculate_smds.py

It goes through the ROI images and calculates the SMDs functions in a dictionary and saves it in an output folder. This part takes a bit of time (8 hours in my pc). so I saved all the smds. This way, we can do some other calculations later if we want.

3) calculate_smd_sum.py

This goes through the pickle files (smds functions saved in step 2) and calculates the sum of first n points (default  =50) and saves the sum in a csv file in an output path. You will also have a column indicating the core id (CS_5057_5_B_1_1_1) so you can merge it later in a data spread sheets. Other colunms are your colunms you already had in "Dataset_BA1B.csv". (likePnS2_sum etc).

hamediut committed:
add cpp file for 512 images, add functions for calculating smds

hamediut committed:
add ROI image selection code

hamediut committed:
add the script to calculate SMDs from ROI core images

hamediut committed:
add usuage in the docstring in calculate_smds.py

hamediut committed:
add a script to calculate sum of smds from smd function calculated by calculate_smds.py

mnky9800n commented:
@hamediut you need to add all this to the `run_pipeline.sh` bash script so that the pipeline can be run and calculate the SMDs or be toggled so that it doesn't run if it will take a long time and they already been calculated.

mnky9800n reviewed (changes requested):
there doesn't seem to be any main integration into the main pipeline via run_pipeline.sh. this should be integrated and tested. @hamediut you need to add all this to the `run_pipeline.sh` bash script so that the pipeline can be run and calculate the SMDs or be toggled so that it doesn't run if it will take a long time and they already been calculated.

hamediut commented:
The issue is that if we put smd calculations in the pipeline, the results would be different than in the paper because ROI selection from the core images are now different than what Elliot did. We already have smd sum values in 'Dataset_BA1B.xlsx' to run meow.py and paperplots.py.
I think the pipeline should include the smd calculation. Instead, we can assume that it is already calculated and put in the colunms of 'Dataset_BA1B.xlsx'. Then, we can explain in the github README.md that if someone wants to do it from scratch or does it on their own images, the smd scripts (that I mentioned above) should be run first. I can then explain step by step how to run each to get smd values.

hamediut committed:
update the smd sum code to save the results in a csv file in the Dataset folder

mnky9800n commented:
That works for me. Please make the commits so that it is represented in this pull request.

hamediut committed:
merge the results of smd in the existing Dataset spreadsheet and updayte bash script

hamediut committed:
update README.md, add the folder to all core images

hamediut committed:
update readme, add ROI selection

hamediut committed:
add different steps of SMD calculation in README.md

hamediut committed:
update readme

hamediut committed:
add tifffile package to the requirements.txt

hamediut committed:
save the output as a new xlsx file instead of replacing the original file from which we want to reproduce the paper results

hamediut commented:
> That works for me. Please make the commits so that it is represented in this pull request.

I have made some changes in the branch. Briefly:
1) Updated the README.md that explain different steps (check and see if it's easy to follow)
2) added one more packages to requirements.txt
3) updated the smd calculation: if you follow the steps, it will create a new dataset called 'Dataset_BA1B_updated.xlsx'. I did this to nor overwrite the original dataset spreadsheet as we want that to be reproducible. I think the better way is to ask for the full path to the exel datasheet to meow.py instead of hardcoding it in you script. Anyway, I think it is almost done, you can then update run_pipeline.sh as you wantI commented the part for calculating smds.
